### DISCUSSION TITLE
Custom Trace ID and Span ID in Open-telemetry Traces

### DISCUSSION THREAD

**Discussion by rdoshi29** [open-telemetry/community#938] [Community Discussions]
Is there any way to set custom Trace IDs and Span IDs to Open-telemetry traces?

**Comment by dyladan** [comment]
What language are you using? Some clients like JS implemented an ID generator plugin system, but those plugins only have limited info about the trace. What sort of custom id are you looking to set?

**Reply by rdoshi29** [reply]
I am using java and I want to set like already existing reference numbers to traces so that I can easily link multiple spans from remote procedures without http headers context propagation.

**Reply by dyladan** [reply]
I don't think there is a way to set a specific trace id, but you might have more luck asking in the java repository at https://github.com/open-telemetry/opentelemetry-java

**Comment by Cinschif** [comment]
I´m using angular into frontend and c# into back. I would like to correlation the trace cata from both, with the same trace id. I´m founding some problems to get it. Is there any way to set custom Trace IDs and Span IDs to Open-telemetry traces? Can i do it using only propagator or there is another way?

**Reply by trask** [reply]
it's generally much easier if you can use the standard W3C `traceparent` header, then you can use the standard W3C TraceContext propagators, and you don't need to set custom trace/span ids

it's possible to set custom trace/span ids, but I'd probably recommend asking that more the language-specific repository that you are working with (e.g. https://github.com/open-telemetry/opentelemetry-js and/or https://github.com/open-telemetry/opentelemetry-dotnet)

**Reply by svrnm** [reply]
> I´m founding some problems to get it.

What kind of problems are you facing? Can you maybe share some sample code and point out what is not working as expected?
